### PR TITLE
Send client and java version to server

### PR DESCRIPTION
**Pull request by tellet-q**
* Add additional info on client and java version to user-agent: `"user-agent": "java-client/v1.12.0 java/21.0.5 grpc-java-netty/1.56.0"`
* Set user-agent during channel creation

### CODE REVIEW

File: src/main/java/io/qdrant/client/QdrantGrpcClient.java [hunk 1]
```diff
@@ -174,10 +177,26 @@ public static class Builder {
...
       this.shutdownChannelOnClose = true;
     }
 
+    private String getClientVersion() {
+      String clientVersion = "Unknown";
+      try {
+        Manifest manifest =
+            new Manifest(Builder.class.getResourceAsStream("/META-INF/MANIFEST.MF"));
+        Attributes attributes = manifest.getMainAttributes();
+        clientVersion = attributes.getValue("X-Qdrant-Version");
```
Comment (Anush008) [inline src/main/java/io/qdrant/client/QdrantGrpcClient.java:193]: This reads from https://github.com/qdrant/java-client/blob/d71b260cd52173822bfc05d064f299692bd74162/build.gradle#L73

Which is the version of the proto files we've downloaded.
We should get `packageVersion` instead. https://github.com/qdrant/java-client/blob/d71b260cd52173822bfc05d064f299692bd74162/gradle.properties#L7-L8

Comment (tellet-q) [reply src/main/java/io/qdrant/client/QdrantGrpcClient.java]: I'll extend manifest with X-Client-Version then.

File: build.gradle [hunk 2]
```diff
@@ -74,6 +74,7 @@ jar {
 		// add git revision and commit time to jar manifest
 		jar.manifest.attributes['X-Git-Revision'] = git.head().id
 		jar.manifest.attributes['X-Git-Commit-Time'] = git.head().dateTime.withZoneSameLocal(ZoneOffset.UTC)
+		jar.manifest.attributes['X-Client-Version'] = project.version
```
Comment (Anush008) [inline build.gradle:77]:
```suggestion
		jar.manifest.attributes['Implementation-Version'] = packageVersion
```

File: src/main/java/io/qdrant/client/QdrantGrpcClient.java [hunk 3]
```diff
@@ -174,10 +177,26 @@ public static class Builder {
...
+      try {
+        Manifest manifest =
+            new Manifest(Builder.class.getResourceAsStream("/META-INF/MANIFEST.MF"));
+        Attributes attributes = manifest.getMainAttributes();
+        clientVersion = attributes.getValue("X-Client-Version");
+      } catch (IOException e) {
+        logger.warn("Failed to read client version from manifest", e);
+      }
+      return clientVersion;
+    }
```
Comment (Anush008) [inline src/main/java/io/qdrant/client/QdrantGrpcClient.java:198]:
```suggestion
```
Won't be needed.

**Review by Anush008 — APPROVED** [review]
🫡

**Pull request merged**